custom-plugin-decorators/azure-apim/README.md

```diff
@@ -0,0 +1,136 @@
+# Azure APIM decorator
+
+Authors:
+
+- [`@adamaltman`](https://github.com/adamaltman), Adam Altman (Redocly)
+
+## What this does and why
+
+[Azure APIM](https://learn.microsoft.com/en-us/azure/api-management/api-management-api-import-restrictions) doesn't support a variety of OpenAPI features (one of which is examples).
+Examples causes an error when attempting to import OpenAPI descriptions that contain examples to Azure APIM.
+
+This decorator removes examples from schemas, media types, and components.
+
+This is a decorator you can run in a pipeline to transform your OpenAPI description prior to uploading to Azure APIM.
+Note that Azure APIM has many restrictions, and it might require more transformation prior to uploading to Azure APIM.
+
+You certainly would want to have the examples for purposes of documentation. Redocly renders multiple media type examples beautifully.
+
+Contribute to this community cookbook by adding decorators for issues you find with your imports to Azure APIM.
+
+## Code
+
+The code is entirely in `azure-apim.js`:
+
+```javascript
+module.exports = {
+    id: "azure-apim",
+    decorators: {
+      oas3: {
+        "remove-examples": RemoveExamples,
+      },
+    },
+  };
+  
+  /** @type {import('@redocly/cli').OasDecorator} */
+  function RemoveExamples() {
+    return {
+      Schema: {
+        leave(Schema) {
+          if (Schema['examples']) {
+            delete Schema['examples'];
+          }
+        }
+      },
+      MediaType: {
+        leave(MediaType) {
+            if (MediaType['examples']) {
+                delete MediaType['examples'];
+            }
+        }
+      },
+      Components: {
+        leave(Components) {
+          if (Components['examples']) {
+            delete Components['examples'];
+          }
+        }
+      }
+    }
+  };
+
+```
```

Comment:
Do we need the duplication? 

```suggestion
The code is entirely in [azure-apim.js](./azure-apim.js).

```

I'm wondering if it really adds value here. For me, copying something from the actual code into a doc makes sense only partially, for example, to explain an essential or tricky fragment.

Reply:
The other decorator readme's have that. I agree it seems unnecessary and I have found cases where they diverge.